### PR TITLE
No scripts

### PR DESCRIPTION
**Pull request by paul-krohn**
Allow ve-packager to skip installing scripts/entry points from all submodules, and also provide a way to call a shim script, in case one needs to do unnatural things to the build before packaging.

Also remove a stray circle.yml file that does not seem to want to go away.

Also too added linewise output to the "pip install -r" step, which can take a spell.

### CODE REVIEW

File: vep/__init__.py [hunk 1]
```diff
@@ -6,6 +6,11 @@
 import sys
 
 
+# pass-through function to enable line-wise output from commands called via sh
+def print_line(line):
+    print(line)
```
Comment (plathrop) [inline vep/__init__.py]: This kinda makes my brain hurt, so I'd really like to hear what made you write this code :-)

File: vep/__init__.py [hunk 2]
```diff
@@ -25,53 +30,68 @@ def add_cli_arguments(self, parser):
...
-            help = "The package version."
+            default=pycmd('setup.py', '--version').strip(),
+            help="The package version."
+        )
+
+        group.add_argument(
+            '--skip-scripts',
+            default=False,
+            action='store_true',
+            help="Set this to skip installing all the scripts in all the setup.py files in all the requirements"
```
Comment (plathrop) [inline vep/__init__.py]: wording nit: most unix commands would phrase this as "Skip installing all the..."

File: vep/__init__.py [hunk 3]
```diff
@@ -25,53 +30,68 @@ def add_cli_arguments(self, parser):
...
+        )
+
+        group.add_argument(
+            '--shim-script',
+            default=None,
+            help="An extra script to run between the build and package steps. "
+                 "If you need to do unnatural things to make your package work, this is the place to do them. "
+                 "Called scripts need to have a shebang line."
```
Comment (plathrop) [inline vep/__init__.py]: This should probably be a little clearer. I assume from above comments that the scripts are executed with `sh`, so maybe just say that in the help?

File: vep/__init__.py [hunk 4]
```diff
@@ -110,8 +130,9 @@ def symlink_entry_points(self):
     def package(self):
         os.chdir(self.args.directory)
         fpm = sh.Command("fpm")
-        print fpm('--verbose', '-s', 'dir', '-t', 'deb', '-n', self.args.package_name, '--prefix', self.args.package_prefix,
-                  '-v', self.args.package_version, '--url', self.args.repo_url, '-C', os.path.join(self.args.directory, self.build_dir), '.')
+        print fpm('--verbose', '-s', 'dir', '-t', 'deb', '-n', self.args.package_name, '--prefix',
+                  self.args.package_prefix, '-v', self.args.package_version, '--url', self.args.repo_url,
+                  '-C', os.path.join(self.args.directory, self.build_dir), '.')
```
Comment (plathrop) [inline vep/__init__.py]: Probably worth comments as to why we chose this specific set of CLI args.

File: vep/__init__.py [hunk 5]
```diff
@@ -142,13 +163,27 @@ def run(self):
         # if there is a requirements.pip, go ahead and install all the things
         if os.path.isfile(self.args.pip_requirements):
             print "installing requirements"
-            print target_pip('install', '-r', self.args.pip_requirements, '-I')
+            # installing requirements can take a spell, print output line-wise
+            target_pip('install', '-r', self.args.pip_requirements, '-I', _out=print_line)
```
Comment (plathrop) [inline vep/__init__.py]: Pretty sure you can just do
```python
_out=print
```
and eliminate `print_line` altogether.

Comment (paul-krohn) [inline vep/__init__.py]: Sadly: 

    $ python
    Python 2.7.10 (default, Jul 14 2015, 19:46:27) 
    [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)] on darwin
    Type "help", "copyright", "credits" or "license" for more information.
    >>> import sh
    >>> ls = sh.Command("ls")
    >>> ls(_out=print)
      File "<stdin>", line 1
        ls(_out=print)
                    ^
    SyntaxError: invalid syntax
    >>> 

Hence the brain-injuring wrapper.

Comment (paul-krohn) [inline vep/__init__.py]: Wait, this is the print-is-not-a-function thing, I can fix that.

**Pull request merged**